### PR TITLE
test: use btrfs by default on Arch as well

### PR DESCRIPTION
**Pull request by mrc0mmand**
So we get coverage for https://github.com/systemd/systemd/pull/31311.

### CODE REVIEW

**Review by bluca — APPROVED** [review]

File: test/TEST-55-OOMD/test.sh [hunk 1]
```diff
@@ -16,6 +16,11 @@ test_append_files() {
     (
         image_install mkswap swapon swapoff stress
 
+        if [[ "$FSTYPE" == btrfs ]]; then
+            # swapfile must not have CoW set, but the flag can be set/unset only on new or empty files
+            touch "${initdir:?}/swapfile"
+            chattr +C "${initdir:?}/swapfile"
+        fi
         dd if=/dev/zero of="${initdir:?}/swapfile" bs=1M count=48
         chmod 0600 "${initdir:?}/swapfile"
```
Comment (YHNdnzj) [inline test/TEST-55-OOMD/test.sh]: Hmm, just use `btrfs filesystem mkswapfile` on btrfs?

File: test/test-functions [hunk 2]
```diff
@@ -1850,7 +1850,7 @@ save_journal() {
```
Comment (yuwata) [inline test/test-functions:1850]: commit message??

**Review by yuwata — APPROVED** [review]

File: test/test-functions (hunk 2 above)
Comment (mrc0mmand) [reply test/test-functions:1850]: Wow, I have no idea how that happened.

**Comment by yuwata** [comment]
Unfortunately, CIs are red.

**Comment by mrc0mmand** [comment]
> Unfortunately, CIs are red.

That's all caused by https://github.com/systemd/systemd/issues/31361, I'll rebase once https://github.com/systemd/systemd/pull/31369 lands, just to be sure.

**Pull request merged**